### PR TITLE
Update text cleaning

### LINKED ISSUE #40: Improve labelling preparation + text cleaning
- [x] Pad character/ number combos
- [ ] Update labelling guide to align to latest sub-categories in the keyword lookup
- [x] Finalise keyword lookup: include social support and cohesion? Shift patterns?

### PR DESCRIPTION
**Pull request by RFOxbury**
---

# Description

This PR modifies the text cleaning slightly. The aim is to be extra vigilant about splitting up things that could potentially be separate sentences, so that these can be tagged separately in the labelling task - see attached screenshots. The first one shows an example from the current labelling task, and the second one shows the same example with the proposed text cleaning changes.

Screenshot 2024-05-02 at 16 26 41
Screenshot 2024-05-02 at 16 22 13

Fixes #40 

# Instructions for Reviewer

In order to test the code in this PR you need to

- [ ] run and inspect the test file `test_text_cleaning.py`

Please pay special attention to ...

# Checklist:

- [x] I have refactored my code out from `notebooks/`
- [x] I have checked the code runs
- [x] I have tested the code
- [x] I have run `pre-commit` and addressed any issues not automatically fixed
- [x] I have merged any new changes from `dev`
- [x] I have documented the code
  - [x] Major functions have docstrings
  - [x] Appropriate information has been added to `README`s
- [x] I have explained this PR above
- [x] I have requested a code review

### CODE REVIEW

File: dap_job_quality/tests/test_text_cleaning.py [hunk 1]
```diff
@@ -0,0 +1,46 @@
+import pytest
```
Comment (india-kerle) [inline dap_job_quality/tests/test_text_cleaning.py:1]: omg a test i'm screaming amazing

Comment (india-kerle) [reply dap_job_quality/tests/test_text_cleaning.py:1]: just ran the test and it passed all tests.

File: dap_job_quality/utils/text_cleaning.py [hunk 2]
```diff
@@ -84,6 +84,19 @@ def detect_camelcase(text):
 }
 
 
+def split_on_period_space(text):
+    """
+    A spaCy pipeline component that inserts a space after a period if the period is followed by an uppercase letter or digit
+    without any intervening space, as long it is not preceded by a digit or the pound sign.
```
Comment (india-kerle) [inline dap_job_quality/utils/text_cleaning.py]: this function doesn't look like it relies on a spacy component? I would update the docstring.

**Review by india-kerle — COMMENTED** [review]
Looks great! I was able to run the tests and they worked a charm. I couldn't think of any edge cases that this might be introducing so i think its fine that the cleaning pipeline looks slightly different for job quality. We can keep an eye in the labelling party. I'd update the docstring for the function to be more accurate but its not necessary for the merge.

File: dap_job_quality/utils/text_cleaning.py (hunk 2 above)
Comment (RFOxbury) [reply dap_job_quality/utils/text_cleaning.py]: Good catch, thank you!
